utils/build-validate-quickstart-artifact.ts

```diff
@@ -0,0 +1,76 @@
+import Quickstart from "./lib/Quickstart";
+import DataSource from "./lib/DataSource";
+import Alert from "./lib/Alert";
+import Dashboard from "./lib/Dashboard";
+import Ajv, { type ErrorObject } from 'ajv';
+
+const main = () => {
+  const schema = require('./schema/artifact.json');
+
+  // 1. Fetch all quickstarts, datasources, alerts, and dashboards
+  const quickstarts = Quickstart.getAll().map((quickstart) => quickstart.config);
+  const dataSources = DataSource.getAll().map((dataSource) => dataSource?.config);
+  const alerts = Alert.getAll().map((alert) => alert.config);
+  const dashboards = Dashboard.getAll().map((dashboard) => dashboard.config);
+
+  const coreDataSourceIds = require('./schema/core-datasource-ids.json');
+  const communityDataSourceIds = dataSources.map((ds) => { return ds?.id });
+
+  // TODO: consider json-schema-to-ts package to infer type from JSON schema
+  //  2. Create the artifact
+  const artifact = {
+    quickstarts,
+    dataSources,
+    alerts,
+    dashboards,
+    dataSourceIds: [...coreDataSourceIds, ...communityDataSourceIds]
+  };
+
+  // 3. Validate the artifact
+  const ajv = new Ajv();
+  ajv.validate(schema, artifact);
+
+  if (ajv.errors?.length) {
+    console.error('*** Validation failed. See errors below. ***');
+    console.error('--------------------------------------------');
+    parseErrors(ajv.errors, artifact);
+
+    process.exit(1);
+  }
+}
+
+const parseErrors = (errors: ErrorObject[], artifact: Record<string, any>) => {
```

Comment:
Down to refactor this if folks feel like it's necessary.

Reply:
Nitpick: could we use `unknown` over `any`? It's slightly less permissive and I think it better signals intent here.